sdk/src/main/java/io/radar/sdk/RadarApiClient.kt

```diff
@@ -949,4 +953,61 @@ internal class RadarApiClient(
         })
     }
 
+    internal fun sendEvent(
+        name: String,
+        metadata: JSONObject?,
+        user: RadarUser?,
+        callback: RadarSendEventApiCallback
+    ) {
+        val publishableKey = RadarSettings.getPublishableKey(context)
+        if (publishableKey == null) {
+            callback.onComplete(RadarStatus.ERROR_PUBLISHABLE_KEY)
+
+            return
+        }
+
+        val params = JSONObject()
+        try {
+            params.putOpt("id", RadarSettings.getId(context))
+            params.putOpt("installId", RadarSettings.getInstallId(context))
+            params.putOpt("userId", RadarSettings.getUserId(context))
+            params.putOpt("deviceId", RadarUtils.getDeviceId(context))
+
+            params.putOpt("type", name)
+            params.putOpt("metadata", metadata)
+
+        } catch (e: JSONException) {
+            callback?.onComplete(RadarStatus.ERROR_BAD_REQUEST)
+
+            return
+        }
+
+        val host = RadarSettings.getHost(context)
+        val uri = Uri.parse(host).buildUpon().appendEncodedPath("v1/events").build()
+        val url = URL(uri.toString())
+
+        val headers = headers(publishableKey)
+        apiHelper.request(context, "POST", url, headers, params, false, object : RadarApiHelper.RadarApiCallback {
+            override fun onComplete(status: RadarStatus, res: JSONObject?) {
+                if (status != RadarStatus.SUCCESS || res == null) {
+                    callback.onComplete(status)
+
+                    return
+                }
+
+                val event = res.optJSONObject("event")?.let { eventObj ->
+                    RadarEvent.fromJson(eventObj)
+                }
+
```

Comment:
this is where I think the logic should happen, and then we can change the callback type to an array of `RadarEvent` for the last arg

Reply:
Yeah would be nice to have that logic here so it's the same as iOS

Reply:
I put it at the top level because the Android version of `RadarApiClient.sendEvent()` doesn't take the list of `trackEvents` like the iOS version does. It probably _should_, but I was trying to minimize the number of changes.

Reply:
it definitely should - that was the modification I made on iOS to do the event coalescing in the API client, let's do that here too